install_sct

```diff
@@ -183,6 +183,28 @@ function check_requirements() {
   if [[ ! ( $(command -v curl) || $(command -v wget) ) ]]; then
     die "ERROR: neither \"curl\" nor \"wget\" is installed. Please install either of them and restart SCT installation."
   fi
+  # check rosetta
+  if [[ "$OS" == "osx" ]]; then
+    if ! arch -x86_64 true >/dev/null 2>&1; then
+      print warning "WARNING: not running an x86_64 architecture."
+      while [[ ! "$ROSETTA_INSTALL" =~ ^([Yy](es)?|[Nn]o?)$ ]]; do
+        ROSETTA_INSTALL="no"
+        if [ -z "$NONINTERACTIVE" ]; then
+          print question "Do you want to install \"Rosetta 2\" now? [y]es/[n]o:"
+          read -r ROSETTA_INSTALL
+        fi
+      done
+      if [[ "$ROSETTA_INSTALL" =~ ^[Yy](es)? ]]; then
+        softwareupdate --install-rosetta
+        # check if we can now run x86_64 executables
+        if ! arch -x86_64 true >/dev/null 2>&1; then
+          die "ERROR: still cannot run x86_64 executables. Please contact SCT team for assistance."
+        fi
+      else
+        die "Please install \"Rosetta 2\" by running \"softwareupdate --install-rosetta\" and restart SCT installation."
+      fi
+    fi
+  fi
```

Comment:
I don't have an Apple Silicon Mac to check, but I do notice that the structure of the addition is [more or less identical](https://www.diffchecker.com/M4Lb416P/) to our existing `gcc` check, with the main functional difference being `arch -x86_64 true` over `gcc --version`.

As you've explained, `arch` on macOS can be used to ["run selected architecture of a universal binary"](https://www.unix.com/man-page/osx/1/arch/). But, the use of [`true`](https://man7.org/linux/man-pages/man1/true.1.html) ("do nothing, successfully") was new to me! Neat. :)

So, `arch -x86_64 true` will fail on Macs without Rosetta 2? :thinking:

Reply:
While researching this, I came across [this SE answer](https://apple.stackexchange.com/a/464165), which suggests both:

```bash
    if ! (arch -arch x86_64 uname -m > /dev/null) ; then
```

Which I guess is more or less identical to `true` anyway, since we're just testing success/failure.

And: 

```bash
softwareupdate --install-rosetta --agree-to-license
```

The behavior comparing "`--agree-to-license` vs. nothing" seems to be documented [here](https://discussions.apple.com/thread/253378816?sortBy=best). tl;dr it looks like it's just a single CLI prompt. 

Given that:

- We're already skipping the Rosetta installation for non-interactive installations anyway
- For interactive installations, the prompt would happen at the start of the installation (in much the same way our other interactive prompts are)
 
I think it should be fine to exclude the `--agree-to-license` as you've done here, since I don't expect it to be any sort of additional nuisance.

Reply:
> So, `arch -x86_64 true` will fail on Macs without Rosetta 2? 🤔

Yes, it should. I wasn't able to test this exactly, since I was testing this on a laptop which already had Rosetta 2 (and I have no idea how to uninstall it), but instead I tried a different architecture and got:

```
~$ arch -x86_64h true
arch: posix-spawnp: true: Bad CPU type in executable
```

> I think it should be fine to exclude the `--agree-to-license` as you've done here, since I don't expect it to be any sort of additional nuisance.

Right. I think the spirit of `--agree-to-license` is that it's for sysadmins running deployment scripts on machines they control, so that they're empowered to agree to the license. In the case of the SCT installation script, I think we (the developers) can't agree to the license on behalf the the user, so we shouldn't hide it.

Reply:
Perfect!